node/src/common/JXCoreEngine.cpp

```diff
@@ -0,0 +1,456 @@
+
+#include <cstdlib>
+#include <functional>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <string>
+#include <thread>
+#include <unordered_map>
+
+#include "Log.h"
+#include "INodeEngine.h"
+#include "AsyncQueue.h"
+#include "WorkItemDispatcher.h"
+#include "JXCoreEngine.h"
+
+#include "jxcore/jx.h"
```

Comment:
Shouldn't we move JXCoreEngine.cpp inside node/src/external/jxcore, instead of having in common dir ?